monte_carlo.py

```diff
@@ -23,60 +27,229 @@
+            batch to be tested
 '''
-def transport_neutron(sigma_t, sigma_s, x_min, x_max, y_min, y_max, z_min, 
-    z_max, crow_distances):
-    neutron_starting_point = list(sample_location(x_min, x_max, y_min, 
-        y_max, z_min, z_max))
-    neutron_position = [neutron_starting_point[i] for i in range(3)] 
+def transport_neutron(mat, bounds, tallies, fission_banks, first_round):
+    
+    '''
+    for i in xrange(10):
+    
+        old_fission_locations.clear()
+        for num in xrange(new_fission_locations.length):
+            old_fission_locations.add(new_fission_locations.next())
+        new_fission_locations.clear()
+    '''
+
+    # first round
+    if first_round:
+        neutron_starting_point = sample_location(bounds)
+        
+    # every other round
+    else:
+        neutron_starting_point = sample_fission_site(fission_banks["old"])
+        
+    neutron_lost = False
+    neutron_reflected = False
+    neutron_position = coords.Coords(neutron_starting_point.x,
+            neutron_starting_point.y, neutron_starting_point.z) 
     neutron_interaction = 0
     while neutron_interaction != 1:
-        neutron_distance = sample_distance(sigma_t)
+        neutron_distance = sample_distance(mat)
         theta = sample_polar_angle()
         phi = sample_azimuthal_angle()
-        neutron_position[0] += neutron_distance*sin(theta)*cos(phi)
-        neutron_position[1] += neutron_distance*sin(theta)*sin(phi)
-        neutron_position[2] += neutron_distance*cos(theta)
-        neutron_interaction = sample_interaction(float(sigma_t),
-            float(sigma_s))
-    neutron_distance_vector = [0, 0, 0]
-    for i in range(3):
-        neutron_distance_vector[i] = neutron_position[i] - \
-            neutron_starting_point[i]
-    crow_distance = sqrt(neutron_distance_vector[0]**2 + \
-        neutron_distance_vector[1]**2 + neutron_distance_vector[2]**2)
-    crow_distances.append(crow_distance) 
-		
+    
+        # neutron movement unit vector
+        neutron_movement_unit_vector = coords.Coords(sin(theta)*cos(phi),
+                   sin(theta)*sin(phi), cos(theta))
+
+        while neutron_distance > 0: 
+            xmaxd = bounds.get_x_max() - neutron_position.x
+            xmind = bounds.get_x_min() - neutron_position.x
+            ymaxd = bounds.get_y_max() - neutron_position.y
+            ymind = bounds.get_y_min() - neutron_position.y
+            zmaxd = bounds.get_z_max() - neutron_position.z
+            zmind = bounds.get_z_min() - neutron_position.z
+               
+            # lim_bound contains the string of the limiting boundary
+            lim_bound = list()
+                
+            # tempd contains the current smallest r
+            tempd = neutron_distance
+                
+            # r is a variable that contains the distance along the
+            # direction vector to the boundary being tested 
+            #test xmax
+            r = xmaxd/(sin(theta)*cos(phi))
```

Comment:
Here is where I think we can remove redundant code since this value is related to neutron_movement_unit_vector